test/unit/local/watcher.js

```diff
@@ -50,6 +51,50 @@ describe('LocalWatcher Tests', function () {
       this.watcher.start()
     })
 
+    it('only recomputes checksums of changed files', async function () {
+      const builders = new MetadataBuilders(this.pouch)
+      const unchangedFilename = 'unchanged-file.txt'
+      const changedFilename = 'changed-file.txt'
+      const unchangedPath = path.join(this.syncPath, unchangedFilename)
+      const changedPath = path.join(this.syncPath, changedFilename)
+      const unchangedData = 'Unchanged file content'
+      const changedData = 'Changed file initial content'
+      await fs.outputFile(unchangedPath, unchangedData)
+      await fs.outputFile(changedPath, changedData)
+      const unchangedStats = await fs.stat(unchangedPath)
+      const {ino: changedIno} = await fs.stat(changedPath)
+      const unchangedDoc = await builders.file()
+        .upToDate()
+        .path(unchangedFilename)
+        .data(unchangedData)
+        .stats(unchangedStats)
+        .create()
+      const changedDoc = await builders.file()
+        .upToDate()
+        .path(changedFilename)
+        .data(changedData)
+        .ino(changedIno)
+        .updatedAt(new Date('2017-03-19T16:44:39.102Z'))
+        .create()
+
+      await fs.outputFile(changedPath, 'Changed file NEW content')
+      this.prep.addFileAsync = sinon.stub().resolves()
+      sinon.spy(this.watcher, 'checksum')
+
+      try {
+        await this.watcher.start()
+
+        should(this.watcher.checksum.args).deepEqual([[changedFilename]])
+        should(await this.pouch.db.get(unchangedDoc._id)).have.properties(unchangedDoc)
+        should(await this.pouch.db.get(changedDoc._id)).have.properties({
+          ...changedDoc
+        })
+      } finally {
+        await this.pouch.db.remove(unchangedDoc)
```

Comment:
Shouldn't the cleanpouch helper takes care of that ?

Reply:
I believe it's in an `after` block, not in an `afterEach` one, and since I lost quite some time fixing unrelated failures because of coupled tests, I didn't want to break everything again...